### PR TITLE
Disable BGS call unless loa3 user

### PR DESCRIPTION
**Pull request by jadon1979**
## Summary

- *Add a check if the user is LOA3 to determine if BGS should be called else return an error*

## Related issue(s)
- [https://github.com/department-of-veterans-affairs/va.gov-team/issues/65772](https://github.com/department-of-veterans-affairs/va.gov-team/issues/65772)

## Testing done

- *Rspec to test for loa3 and non loa3 users*

## What areas of the site does it impact?
*(This impacts calls to the hca rating_info endpoint)*

### CODE REVIEW

File: app/controllers/v0/health_care_applications_controller.rb [hunk 1]
```diff
@@ -80,6 +81,15 @@ def download_pdf
 
     private
 
+    def validate_for_bgs
+      mpi_data = MPIData.new(current_user.identity)
+      response = mpi_data.response_from_redis_or_service(user_key: 'mpi_verification')
```
Comment (lihanli) [inline app/controllers/v0/health_care_applications_controller.rb:86]: you don't need to make a call to MPI. you can just check `current_user.loa3?` and `current_user.ssn.present?`

File: spec/requests/health_care_applications_request_spec.rb [hunk 2]
```diff
@@ -17,17 +17,29 @@
 
     before do
       sign_in_as(current_user)
+      stub_mpi
     end
 
     it 'returns the users rating info' do
-      VCR.use_cassette('bgs/service/find_rating_data', VCR::MATCH_EVERYTHING) do
```
Comment (lihanli) [inline spec/requests/health_care_applications_request_spec.rb:23]: why was `VCR::MATCH_EVERYTHING` deleted?

File: spec/requests/health_care_applications_request_spec.rb [hunk 3]
```diff
@@ -17,17 +17,29 @@
...
         get(rating_info_v0_health_care_applications_path)
       end
 
       expect(JSON.parse(response.body)['data']['attributes']).to eq(
         { 'user_percent_of_disability' => 100 }
       )
     end
+
+    it 'errors if user is not loa3' do
+      VCR.use_cassette('bgs/service/find_rating_data') do
```
Comment (lihanli) [inline spec/requests/health_care_applications_request_spec.rb:34]: you don't need to use VCR here, it's supposed to block the user from the API and it shouldn't be making a call to BGS

**Review by JoshingYou1 — APPROVED** [review]
LGTM! Nice work Chapley! 😄

File: app/policies/hca_disability_rating_policy.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+HCADisabilityRatingPolicy = Struct.new(:user, :form1095) do
```
Comment (lihanli) [inline app/policies/hca_disability_rating_policy.rb:3]: :form1095 should be changed to :hca_disability_rating

File: spec/requests/health_care_applications_request_spec.rb [hunk 5]
```diff
@@ -28,6 +28,15 @@
         { 'user_percent_of_disability' => 100 }
       )
     end
+
+    it 'errors if user is not loa3' do
+      build(:user, :loa1)
```
Comment (lihanli) [inline spec/requests/health_care_applications_request_spec.rb:33]: delete this line and write the spec like this
```ruby
context 'with an loa1 user' do
  let(:current_user) { build(:user, :loa1) }

  it 'errors if user is not loa3' do
     # ...
  end
end
```

**Review by lihanli — CHANGES_REQUESTED** [review]

**Review by lihanli — DISMISSED** [review]
thanks for making the changes

**Review by JoshingYou1 — DISMISSED** [review]

**Review by lihanli — APPROVED** [review]

**Pull request merged**